dali/operators/reader/video_reader_decoder_op.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "dali/operators/reader/video_reader_decoder_op.h"
+
+#include <string>
+#include <vector>
+
+namespace dali {
+
+VideoReaderDecoder::VideoReaderDecoder(const OpSpec &spec)
+    : DataReader<CPUBackend, VideoSample>(spec),
+      has_labels_(spec.HasArgument("labels")) {
+      loader_ = InitLoader<VideoLoaderDecoder>(spec);
+}
+
+void VideoReaderDecoder::RunImpl(SampleWorkspace &ws) {
+  const auto &sample = GetSample(ws.data_idx());
+  auto &video_output = ws.Output<CPUBackend>(0);
+
+  video_output.Copy(sample.data_, 0);
+
+  if (has_labels_) {
+    auto &label_output = ws.Output<CPUBackend>(1);
+    label_output.Resize({});
+    label_output.set_type<float>();
```

Comment:
I think that there's an API to set both the shape and the type at once, which if I remember correctly is more efficient.

Reply:
Done